pkg/catalog/routes_test.go

```diff
@@ -937,6 +937,165 @@ func TestBuildInboundPoliciesSameNamespace(t *testing.T) {
 	assert.ElementsMatch(expectedPolicies, actual)
 }
 
+func TestBuildInboundAllowAllPolicies(t *testing.T) {
+	assert := tassert.New(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockKubeController := k8s.NewMockController(mockCtrl)
+	mockMeshSpec := smi.NewMockMeshSpec(mockCtrl)
+	mockEndpointProvider := endpoint.NewMockProvider(mockCtrl)
+
+	mc := MeshCatalog{
+		kubeController:     mockKubeController,
+		meshSpec:           mockMeshSpec,
+		endpointsProviders: []endpoint.Provider{mockEndpointProvider},
+	}
+
+	bookstoreSA := service.K8sServiceAccount{
+		Name:      "bookstore",
+		Namespace: "bookstore-ns",
+	}
+
+	bookbuyerSA := service.K8sServiceAccount{
+		Name:      "bookbuyer",
+		Namespace: "bookbuyer-ns",
+	}
+	bookstoreMeshService := service.MeshService{
+		Name:      "bookstore",
+		Namespace: "bookstore-ns",
+	}
+	bookstoreK8sService := tests.NewServiceFixture(bookstoreMeshService.Name, bookstoreMeshService.Namespace, map[string]string{})
+
+	serviceAccounts := []*corev1.ServiceAccount{}
+	serviceAccounts = append(serviceAccounts, tests.NewServiceAccountFixture(tests.BookbuyerServiceAccountName, "bookbuyer-ns"))
+	serviceAccounts = append(serviceAccounts, tests.NewServiceAccountFixture(tests.BookstoreServiceAccountName, "bookstore-ns"))
+
+	mockEndpointProvider.EXPECT().GetID().Return("fake").AnyTimes()
```

Comment:
It would be nice to convert this to a table driven test for extending this test case to new input scenarios. Would you mind doing so, a separate PR for this works.

Reply:
I have updated the current PR itself to incorporate table driven test